### PR TITLE
Added in template stuff for t1493

### PR DESCRIPTION
**Pull request by k88hudson**

### CODE REVIEW

File: templates/supported/shared/plugins/flickr/popcorn.flickr.html [hunk 1]
```diff
@@ -0,0 +1,76 @@
+<!DOCTYPE html>
```
Comment (k88hudson) [inline templates/supported/shared/plugins/flickr/popcorn.flickr.html]: No changes here from master

File: templates/supported/shared/plugins/flickr/popcorn.flickr.js [hunk 2]
```diff
@@ -0,0 +1,256 @@
+// PLUGIN: Flickr
+(function (Popcorn) {
```
Comment (k88hudson) [inline templates/supported/shared/plugins/flickr/popcorn.flickr.js:2]: Only a couple changes marked inline...

File: templates/supported/shared/plugins/flickr/popcorn.flickr.js [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+        _border = options.border || "0px";
+
+    // create a new div this way anything in the target div is left intact
+    // this is later populated with Flickr images
+    containerDiv = document.createElement( "div" );
+    containerDiv.id = "flickr" + idx;
+    containerDiv.style.width = "100%";
+    containerDiv.style.height = "100%";
+    containerDiv.style.display = "none";
+    containerDiv.classList.add( "popcorn-flickr" );
```
Comment (k88hudson) [inline templates/supported/shared/plugins/flickr/popcorn.flickr.js:66]: Change here: add a class so I can select with css

File: templates/supported/shared/plugins/flickr/popcorn.flickr.js [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+        label: "Api_key",
+        optional: true,
+        hidden: true
+      },
+      target: "flickr-container",
+      height: {
+        elem: "input",
+        type: "text",
+        label: "Height",
+        "default": "200px",
```
Comment (k88hudson) [inline templates/supported/shared/plugins/flickr/popcorn.flickr.js]: change here, need a different default

File: templates/supported/shared/plugins/flickr/popcorn.flickr.js [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+        optional: true,
+        hidden: true
+      },
+      width: {
+        elem: "input",
+        type: "text",
+        label: "Width",
+        "default": "200px",
```
Comment (k88hudson) [inline templates/supported/shared/plugins/flickr/popcorn.flickr.js]: same as above

File: templates/supported/shared/plugins/flickr/popcorn.flickr.js [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+        type: "text",
+        label: "Border",
+        "default": "0",
+        optional: true,
+        hidden: true
+      },
+      numberofimages: {
+        elem: "input",
+        type: "number",
+        "default": 10,
```
Comment (k88hudson) [inline templates/supported/shared/plugins/flickr/popcorn.flickr.js:247]: same as above

File: templates/supported/shared/plugins/flickr/popcorn.flickr.unit.html [hunk 7]
```diff
@@ -0,0 +1,50 @@
+<!DOCTYPE html>
```
Comment (k88hudson) [inline templates/supported/shared/plugins/flickr/popcorn.flickr.unit.html]: No changes in this file

File: templates/supported/shared/plugins/flickr/popcorn.flickr.unit.js [hunk 8]
```diff
@@ -0,0 +1,100 @@
+test( "Popcorn Flickr Plugin", function() {
```
Comment (k88hudson) [inline templates/supported/shared/plugins/flickr/popcorn.flickr.unit.js]: Again, no changes

File: templates/supported/shared/plugins/googlemap/popcorn.googlemap.html [hunk 9]
```diff
@@ -0,0 +1,193 @@
```
Comment (k88hudson) [inline templates/supported/shared/plugins/googlemap/popcorn.googlemap.html]: No changes

File: templates/supported/shared/plugins/googlemap/popcorn.googlemap.js [hunk 10]
```diff
@@ -0,0 +1,444 @@
...
+        type: "text",
+        label: "Out"
+      },
+      target: "map-container",
+      type: {
+        elem: "select",
+        options: [ "ROADMAP", "SATELLITE", "STREETVIEW", "HYBRID", "TERRAIN", "STAMEN-WATERCOLOR", "STAMEN-TERRAIN", "STAMEN-TONER" ],
+        label: "Type",
+        optional: true,
+        "default": "STAMEN-TONER"
```
Comment (k88hudson) [inline templates/supported/shared/plugins/googlemap/popcorn.googlemap.js:400]: Changed default for story camp.

File: templates/supported/shared/plugins/googlemap/popcorn.googlemap.js [hunk 11]
```diff
@@ -0,0 +1,444 @@
...
+      },
+      zoom: {
+        elem: "input",
+        type: "text",
+        label: "Zoom",
+        "default": 9,
```
Comment (k88hudson) [inline templates/supported/shared/plugins/googlemap/popcorn.googlemap.js:406]: Changed default for story camp.

File: templates/supported/shared/plugins/googlemap/popcorn.googlemap.unit.html [hunk 12]
```diff
@@ -0,0 +1,57 @@
+<!DOCTYPE html>
```
Comment (k88hudson) [inline templates/supported/shared/plugins/googlemap/popcorn.googlemap.unit.html]: No changes

File: templates/supported/shared/plugins/googlemap/popcorn.googlemap.unit.js [hunk 13]
```diff
@@ -0,0 +1,156 @@
+test( "Popcorn Google Map Plugin", function() {
```
Comment (k88hudson) [inline templates/supported/shared/plugins/googlemap/popcorn.googlemap.unit.js]: No changes

File: templates/supported/shared/plugins/flickr/popcorn.flickr.js (hunk 3 above)
Comment (dseif) [inline templates/supported/shared/plugins/flickr/popcorn.flickr.js:66]: do we have access to our classList shim here? If so this is fine.

File: templates/supported/shared/plugins/flickr/popcorn.flickr.html [hunk 14]
```diff
@@ -0,0 +1,76 @@
...
+  <script src="../../popcorn.js"></script>
+  <script src="popcorn.flickr.js"></script>
+  <script>
+    document.addEventListener( "DOMContentLoaded", function () {
+
+      var p = Popcorn( "#video" )
+      .flickr({
+        start: 0,
+        end: 5,
+        username: "AniaSob",
```
Comment (humphd) [inline templates/supported/shared/plugins/flickr/popcorn.flickr.html]: This terrifies me a bit, with the API key.  Can we somehow leave it out?

Comment (humphd) [inline templates/supported/shared/plugins/flickr/popcorn.flickr.html]: Actually, this file can just get deleted.

File: templates/supported/shared/plugins/flickr/popcorn.flickr.js [hunk 15]
```diff
@@ -0,0 +1,256 @@
...
+    }
+
+    target && target.appendChild( containerDiv );
+
+    // get the userid from Flickr API by using the username and apikey
+    var isUserIDReady = function() {
+      if ( !_userid ) {
+
+        _uri  = "http://api.flickr.com/services/rest/?method=flickr.people.findByUsername&";
+        _uri += "username=" + options.username + "&api_key=" + options.apikey + "&format=json&jsoncallback=flickr";
```
Comment (humphd) [inline templates/supported/shared/plugins/flickr/popcorn.flickr.js:76]: Here, this API key stuff...

File: templates/supported/shared/plugins/flickr/popcorn.flickr.js (hunk 4 above)
Comment (humphd) [inline templates/supported/shared/plugins/flickr/popcorn.flickr.js]: Doesn't this go back on our 200 vs. 200px debate?  I thought we agreed on numbers?

File: templates/supported/shared/plugins/flickr/popcorn.flickr.js (hunk 5 above)
Comment (humphd) [inline templates/supported/shared/plugins/flickr/popcorn.flickr.js]: Same issue here.

File: templates/supported/shared/plugins/flickr/popcorn.flickr.unit.html (hunk 7 above)
Comment (humphd) [inline templates/supported/shared/plugins/flickr/popcorn.flickr.unit.html]: Just delete this file

File: templates/supported/shared/plugins/flickr/popcorn.flickr.unit.js (hunk 8 above)
Comment (humphd) [inline templates/supported/shared/plugins/flickr/popcorn.flickr.unit.js]: Same, delete.

File: templates/supported/shared/plugins/googlemap/popcorn.googlemap.html (hunk 9 above)
Comment (humphd) [inline templates/supported/shared/plugins/googlemap/popcorn.googlemap.html]: Delete.

File: templates/supported/shared/plugins/googlemap/popcorn.googlemap.unit.html (hunk 12 above)
Comment (humphd) [inline templates/supported/shared/plugins/googlemap/popcorn.googlemap.unit.html]: delete.

File: templates/supported/shared/plugins/googlemap/popcorn.googlemap.unit.js (hunk 13 above)
Comment (humphd) [inline templates/supported/shared/plugins/googlemap/popcorn.googlemap.unit.js]: delete.

File: templates/supported/shared/plugins/flickr/popcorn.flickr.js (hunk 15 above)
Comment (dseif) [inline templates/supported/shared/plugins/flickr/popcorn.flickr.js:76]: Will this plugin still work without the api key stuff tho?

File: templates/supported/shared/plugins/titles/popcorn.titles.js [hunk 16]
```diff
@@ -0,0 +1,166 @@
...
+          elem: "input",
+          type: "text",
+          label: "Text",
+          "default": "Popcorn.js"
+        },
+        innerContainerClasses: {
+          elem: "input",
+          type: "text",
+          label: "Inner Container Classes (separate with a space)",
+          "default": ""
```
Comment (dseif) [inline templates/supported/shared/plugins/titles/popcorn.titles.js:58]: We don't need to set empty defaults, we do this in trackevent.js if there is no default

File: templates/supported/shared/plugins/titles/popcorn.titles.js [hunk 17]
```diff
@@ -0,0 +1,166 @@
...
+      // Cache reference to actual target container
+      options._target = target;
+
+
+      // --------------------------------------
+      // PREPARE THE CONTAINER
+
+      // Add each class to the inner container
+      for( i=0; i< innerContainerClasses.length; i++ ){
+        innerContainer.classList.add( innerContainerClasses[i] );
```
Comment (dseif) [inline templates/supported/shared/plugins/titles/popcorn.titles.js:127]: same here, do we have access to the classList shim?

File: templates/supported/shared/plugins/titles/popcorn.titles.js [hunk 18]
```diff
@@ -0,0 +1,166 @@
...
+      }
+
+      // Add each class to the outer container
+      for( j=0; j<containerClasses.length; j++ ){
+        container.classList.add( containerClasses[j] );
```
Comment (dseif) [inline templates/supported/shared/plugins/titles/popcorn.titles.js:132]: same here

File: templates/supported/shared/plugins/titles/popcorn.titles.js [hunk 19]
```diff
@@ -0,0 +1,166 @@
...
+      }
+      // Add transition styles and hide on setup
+      container.classList.add( transitionClass );
+      container.classList.add( inactiveClass );
```
Comment (dseif) [inline templates/supported/shared/plugins/titles/popcorn.titles.js:136]: and here

Comment (dseif) [inline templates/supported/shared/plugins/titles/popcorn.titles.js:136]: and below as well ( sorry for commented so much on the same issue, didn't look down before commenting )

File: templates/supported/shared/plugins/titles/popcorn.titles.js (hunk 17 above)
Comment (humphd) [inline templates/supported/shared/plugins/titles/popcorn.titles.js:127]: classList shim gets rolled into the built butter we'll use via the storycamp make target.

Comment (humphd) [inline templates/supported/shared/plugins/titles/popcorn.titles.js:127]: correction, into the popcorn.js we build, so will work on export too.

Comment (dseif) [inline templates/supported/shared/plugins/titles/popcorn.titles.js:127]: cool, everything I've mentioned about the classList stuff can be ignored then

File: templates/supported/shared/plugins/flickr/popcorn.flickr.js (hunk 4 above)
Comment (k88hudson) [inline templates/supported/shared/plugins/flickr/popcorn.flickr.js]: I filed a ticket to solve this inside popcorn repo:

https://webmademovies.lighthouseapp.com/projects/63272-popcornjs/tickets/1146-make-sure-all-plugins-use-numbers-for-positioning-and-size-instead-of-strings-where-possible

File: templates/supported/shared/plugins/titles/popcorn.titles.js (hunk 18 above)
Comment (k88hudson) [inline templates/supported/shared/plugins/titles/popcorn.titles.js:132]: yes, there are other plugins that use classList

File: templates/supported/shared/plugins/flickr/popcorn.flickr.js (hunk 15 above)
Comment (cadecairos) [inline templates/supported/shared/plugins/flickr/popcorn.flickr.js:76]: Sadly, it won't work without a key.

File: templates/supported/shared/plugins/titles/popcorn.titles.js (hunk 16 above)
Comment (cadecairos) [inline templates/supported/shared/plugins/titles/popcorn.titles.js:58]: This is a part of the text plugin that allows it to circumvent that trackevent.js default stuff.

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @k88hudson received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/80c9b639a731ab5/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/80c9b639a731ab5/output.txt

Total script time: 0.58 mins

Server is running on http://pr475.butyr.org/templates/test.html

**Pull request merged**